src/Effectful/Resource.hs

```diff
@@ -100,6 +106,17 @@ registerEff release = do
     es1 <- cloneEnv es0
     RI.register' istate $ unEff release es1
 
+-- | A variant of 'R.release' adjusted to work in the 'Eff' monad.
+releaseEff :: Resource :> es => R.ReleaseKey -> Eff es ()
+releaseEff = unsafeEff_ . R.release
+
+-- | A variant of 'R.unprotect' adjusted to work in the 'Eff' monad.
+--
+-- /Note:/ the @release@ action returned will run a clone of the environment it
```

Comment:
Will it? I don't know what `unprotect` does, but I don't see `cloneEnv` below (unlike in `allocateEff`).